### DISCUSSION TITLE
starting over in tech.

### DISCUSSION THREAD

**Discussion by Maxsweet1** [Cyfrin/foundry-full-course-cu#2944] [General]
I was  looking for some advice.  I am a new to tech in general and blockchain in particular.  I have had some difficultly grasping some of the concepts in foundry.  I was wondering if i should stop and review or complete the course work and then go back to topics that i had difficulty.   Any advice would be helpful.

**Comment by EngrPips** [comment]
Don't be lazy, and success is inevitable.

**Comment by heheboii11** [comment]
If you feel like you are not able to grasp a certain topic feel free to go back to the topic and do it again, spend some time thinking why it is happening like that and storm your brain for understanding the concept, just search the internet, ask your ai buddy on how the things are working. Question the ai about the part you are not getting, just ask each and everything till you understand how it works. But still if you are not getting how the code is working or what is wrong with your code, just post here in discussions about the error you are facing or the concept you are facing difficult in understanding. 
Even if you are feeling difficult with writing code, just push through it by understanding it. Things will get to unfold as you get through the course. It will be much easier to understand the code, you will understand to debug the code on yourself. Just stayu cosnistent and keeping pushing through the hard times!!!!. 
All the best for your course!!

**Reply by EngrPips** [reply]
Great heads-up.

**Comment by chainflash** [comment]
don't be disappointed, this part is one of the hardest parts of this course.
At any stage, go back and review the previous content and continue again.

**Reply by EngrPips** [reply]
Great heads-up.

**Comment by 0x1057** [comment]
I actually had some issue when I hit the Foundry courses as well.  I went and watched Smart Contract Programmer's Foundry playlist to get my feet under me and from then on I got A TON more out of Patrick's Foundry courses.  YMMV but I would give it a try if you are struggling with that specific part of Updraft.  Link below for your convenience.  

https://www.youtube.com/playlist?list=PLO5VPQH6OWdUrKEWPF07CSuVm3T99DQki

**Reply by EngrPips** [reply]
Great heads-up.

**Reply by KMean** [reply]
@nightfallxyz that's actually really good. Enjoyed it!
